docs/conf.py

```diff
@@ -85,6 +85,16 @@
 # The master toctree document.
 master_doc = "index"
 
+# Ignoring links from ScienceDirect
+linkcheck_ignore = [
+    "https://www.sciencedirect.com",
+    "https://doi.org/10.1137/20M1336898",  # DOI link to ignore
+    "https://en.wikipedia.org/wiki/",  # Wikipedia link to ignore
+    "https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#a-before-you-begin",  # GitHub link to ignore
```

Comment:
Why is this being ignored? It seems weird that we cannot check a link to our own code base. I would rather have a different workaround than ignore this link

Reply:
I think the reason for this was that GitHub URLs are never happy with link checkers and have aggressive rate limits (especially because we can't supply `GITHUB_TOKEN` unlike how we can Lychee)

IMO, it seems fine to ignore because Lychee is checking this URL anyway, but we should add a comment about it somewhere.

Reply:
I still think it is better to find another workaround. You can add secrets to read the docs if that is the problem with an alternative fix

Reply:
No, my point is that it might be better that the Sphinx linkcheck can be reserved to docs-only links, and Lychee can check the rest of the links, which was my idea when I opened #3387.

Adding secrets to the Read the Docs config sounds like another nightmare when everything is moving towards environments and trusted publishing, especially for a GitHub App like it that's already hogging up many permissions when setting it up. So, I wouldn't really recommend that.

At this moment, the Contributing guide is just `..include ::`d into the docs, but it resides outside it in the repository's root (I implemented this change back then), so one reasonable workaround is to replace the GitHub link with the RTD link:

`https://docs.pybamm.org/en/latest/source/user_guide/contributing.html#a-before-you-begin` (note the "latest" string in the URL because it's a development-related document).

Reply:
I like the RTD link idea if it works for both

Reply:
@RohitP2005 Can you do the RTD link change, that is pretty much the last thing I want from the changes